src/openrct2/ride/RideConstruction.cpp

```diff
@@ -1336,37 +1339,42 @@ CoordsXYZD ride_get_entrance_or_exit_position_from_screen_position(const ScreenC
                         continue;
                     if (tileElement->AsTrack()->GetTrackType() == TrackElemType::Maze)
                     {
+                        // if it's a maze, it can place the entrance and exit immediately
                         entranceExitCoords.direction = direction_reverse(entranceExitCoords.direction);
                         gRideEntranceExitPlaceDirection = entranceExitCoords.direction;
                         return entranceExitCoords;
                     }
                     if (tileElement->AsTrack()->GetStationIndex() != gRideEntranceExitPlaceStationIndex)
                         continue;
+                    // if it's not a maze, the sequence properties for the TrackElement must be found to determine if an
+                    // entrance can be placed on that side
 
-                    int32_t eax = (entranceExitCoords.direction + 2 - tileElement->GetDirection())
-                        & TILE_ELEMENT_DIRECTION_MASK;
+                    // get the exit hut side relative to the TrackElement
+                    Direction direction = (direction_reverse(entranceExitCoords.direction) - tileElement->GetDirection()) & 3;
                     const auto& ted = GetTrackElementDescriptor(tileElement->AsTrack()->GetTrackType());
-                    if (ted.SequenceProperties[tileElement->AsTrack()->GetSequenceIndex()] & (1 << eax))
+                    if (ted.SequenceProperties[tileElement->AsTrack()->GetSequenceIndex()] & (1 << direction))
                     {
+                        // if that side of the TrackElement supports stations, the entrance hut is valid and faces away from the
+                        // station
                         entranceExitCoords.direction = direction_reverse(entranceExitCoords.direction);
                         gRideEntranceExitPlaceDirection = entranceExitCoords.direction;
                         return entranceExitCoords;
                     }
                 } while (!(tileElement++)->IsLastForTile());
             }
-            entranceExitCoords.direction = (entranceExitCoords.direction + 1) & 3;
         }
         gRideEntranceExitPlaceDirection = INVALID_DIRECTION;
     }
     else
     {
-        auto mapX = stationStart.x;
-        auto mapY = stationStart.y;
-        auto entranceMinX = mapX;
-        auto entranceMinY = mapY;
-        auto entranceMaxX = mapX;
-        auto entranceMaxY = mapY;
-
+        // create a 2-point rectangle. The rectangle will have a width or height of 0 depending on station direction or length
+        // (if the station is 0) The 2-point rectangle is ballooned on all four sides by 1 tile for a 3xN or Nx3 rectangle.
+        // Whatever edge the entrance hut is on, rotate it outward to get entrance direction.
+        CoordsXY nextLocation = { stationStart.x, stationStart.y };
```

Comment:
```suggestion
        CoordsXY nextLocation = { stationStart };
```
should work, I think.